### PR TITLE
[Performance] Load Video for main screen

### PR DESCRIPTION
**Pull request by andriyParashchuk**
Need load video different quality for a specific device or reduce quality on video

### CODE REVIEW

**Comment by andriyParashchuk** [comment]
Staging - http://jt-pr295.surge.sh/
@pftg @obliviusm @sweatC Pls check

File: _assets/javascripts/application.js [hunk 1]
```diff
@@ -197,7 +197,21 @@ $(document).ready(function () {
...
 
 var font = new FontFaceObserver('Graphik Web');
-
 font.load().then(function () {
   document.documentElement.className += " fonts-loaded";
 });
+
+var mainVideo = document.querySelector(".main-video");
+var videoBg = function() {
+  if (window.innerWidth > 1365) {
```
Comment (pftg) [inline _assets/javascripts/application.js]: you could check is it visible or not

File: _assets/javascripts/application.js [hunk 2]
```diff
@@ -197,7 +197,21 @@ $(document).ready(function () {
...
+  }
+
+  for (var source in mainVideo.children) {
+    var videoSource = mainVideo.children[source];
+    if (typeof videoSource.tagName === "string" && videoSource.tagName === "SOURCE") {
+      videoSource.src = videoSource.dataset.src;
+    }
+  }
+};
+videoBg();
```
Comment (pftg) [inline _assets/javascripts/application.js]: why do you need to name this function?

File: _assets/javascripts/application.js [hunk 3]
```diff
@@ -197,7 +197,21 @@ $(document).ready(function () {
...
+    return false
+  }
+
+  for (var source in mainVideo.children) {
+    var videoSource = mainVideo.children[source];
+    if (typeof videoSource.tagName === "string" && videoSource.tagName === "SOURCE") {
+      videoSource.src = videoSource.dataset.src;
+    }
+  }
+};
```
Comment (pftg) [inline _assets/javascripts/application.js]: what is the purpose of this function? it loads video on html parsed. What is the difference with prev version?

**Review by pftg — CHANGES_REQUESTED** [review]
Not clear with this solution. Could you clarify your strategy?

File: _assets/javascripts/application.js [hunk 4]
```diff
@@ -197,7 +197,21 @@ $(document).ready(function () {
 });
 
 var font = new FontFaceObserver('Graphik Web');
-
 font.load().then(function () {
   document.documentElement.className += " fonts-loaded";
 });
+
+var mainVideo = document.querySelector(".main-video");
```
Comment (pftg) [inline _assets/javascripts/application.js]: why you have not used https://github.com/jetthoughts/jetthoughts.github.io/pull/295/files#diff-2a4fde3d05134dead2ded8c935cda415L171 this style to convert to array?

File: _includes/sections/main-screen.html [hunk 5]
```diff
@@ -1,7 +1,8 @@
 <section class="section main-screen">
   <div class="video-cover mobile-hide">
-    <video playsinline autoplay muted loop preload>
-      <source src="{% asset video_bg.mp4 @path %}" type="video/mp4">
+    <video playsinline autoplay muted loop class="main-video">
+      <source src="{% asset video_bg_full_hd.m4v @path %}" data-src="{% asset video_bg_hd.m4v @path %}" type="video/mp4">
+      <source src="{% asset video_bg_full_hd.mp4 @path %}" data-src="{% asset video_bg_hd.mp4 @path %}" type="video/mp4">
```
Comment (pftg) [inline _includes/sections/main-screen.html]: as I got bg hd should be default, and full hd should be later

**Comment by pftg** [comment]
@andriyParashchuk I just want to use this https://cl.ly/182t190k2D1q/Image%202018-05-03%20at%2003.45.46.public.png

could you check https://developer.mozilla.org/en-US/docs/Web/HTML/Element/source and without polyfill other browsers

**Comment by pftg** [comment]
https://rigor.com/blog/2016/03/size-and-quality-optimizations-for-html5-video

File: _includes/sections/main-screen.html (hunk 5 above)
Comment (andriyParashchuk) [reply _includes/sections/main-screen.html]: Accepted

File: _assets/javascripts/application.js (hunk 4 above)
Comment (andriyParashchuk) [reply _assets/javascripts/application.js]: Accepted

File: _assets/javascripts/application.js (hunk 3 above)
Comment (andriyParashchuk) [reply _assets/javascripts/application.js]: This function checks the screen width
If the screen has a width of more than 1366px, the load Full HD video
If the screen has a width of less than 1366px, the load HD video

File: _assets/javascripts/application.js (hunk 2 above)
Comment (andriyParashchuk) [reply _assets/javascripts/application.js]: Accepted

**Comment by andriyParashchuk** [comment]
this - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/source
Attributes size specifies only the width, we set the width of the video and center it in css
Attribute media is what we need, but it only works with <picture> element

**Comment by andriyParashchuk** [comment]
staging - http://jt-pr295.surge.sh/

**Comment by andriyParashchuk** [comment]
Clean up js code

**Review by pftg — APPROVED** [review]
LGTM

**Pull request merged**